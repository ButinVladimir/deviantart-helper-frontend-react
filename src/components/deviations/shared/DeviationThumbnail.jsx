import React from 'react';
import PropTypes from 'prop-types';
import Media from 'react-bulma-components/lib/components/media';
import Tag from 'react-bulma-components/lib/components/tag';

export default function DeviationThumbnail({
  title,
  thumbnail,
  url,
  publishedTime,
  nsfw,
}) {
  const publishedTimeDate = new Date(publishedTime);

  return (
    <Media.Item position="left" className="image-container">
      <div>
        <img
          alt={title}
          src={thumbnail.src}
        />
      </div>
      <div>
        <small className="has-text-grey">
          {publishedTimeDate.toLocaleString()}
        </small>
      </div>
      <div>
        <a href={url}>Visit DA page</a>
      </div>
      {nsfw && <Tag color="danger">NSFW</Tag>}
    </Media.Item>
  );
}

DeviationThumbnail.propTypes = {
  title: PropTypes.string.isRequired,
  thumbnail: PropTypes.shape({
    src: PropTypes.string.isRequired,
    width: PropTypes.number.isRequired,
    height: PropTypes.number.isRequired,
  }).isRequired,
  url: PropTypes.string.isRequired,
  publishedTime: PropTypes.number.isRequired,
  nsfw: PropTypes.bool.isRequired,
};
