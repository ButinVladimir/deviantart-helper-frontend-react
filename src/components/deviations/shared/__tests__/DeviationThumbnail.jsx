import React from 'react';
import { shallow } from 'enzyme';
import DeviationThumbnail from '../DeviationThumbnail';

describe('DeviationThumbnail', () => {
  const thumbnailProps = {
    title: 'Title',
    thumbnail: {
      src: '/path/to/thumbnail',
      width: 1,
      height: 2,
    },
    url: '/path/to/deviation',
    publishedTime: 1234,
  };

  it('can be rendered correctly when deviation is not NSFW', () => {
    const wrapper = shallow(<DeviationThumbnail {...thumbnailProps} nsfw={false} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('can be rendered correctly when deviation is NSFW', () => {
    const wrapper = shallow(<DeviationThumbnail {...thumbnailProps} nsfw />);

    expect(wrapper).toMatchSnapshot();
  });
});
