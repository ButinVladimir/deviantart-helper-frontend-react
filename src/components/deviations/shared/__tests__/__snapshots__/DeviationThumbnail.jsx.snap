// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviationThumbnail can be rendered correctly when deviation is NSFW 1`] = `
<MediaItem
  className="image-container"
  position="left"
  renderAs="div"
  style={Object {}}
>
  <div>
    <img
      alt="Title"
      src="/path/to/thumbnail"
    />
  </div>
  <div>
    <small
      className="has-text-grey"
    >
      1970-1-1 03:00:01
    </small>
  </div>
  <div>
    <a
      href="/path/to/deviation"
    >
      Visit DA page
    </a>
  </div>
  <Tag
    className=""
    color="danger"
    remove={false}
    renderAs="span"
    rounded={false}
    size={null}
    style={Object {}}
  >
    NSFW
  </Tag>
</MediaItem>
`;

exports[`DeviationThumbnail can be rendered correctly when deviation is not NSFW 1`] = `
<MediaItem
  className="image-container"
  position="left"
  renderAs="div"
  style={Object {}}
>
  <div>
    <img
      alt="Title"
      src="/path/to/thumbnail"
    />
  </div>
  <div>
    <small
      className="has-text-grey"
    >
      1970-1-1 03:00:01
    </small>
  </div>
  <div>
    <a
      href="/path/to/deviation"
    >
      Visit DA page
    </a>
  </div>
</MediaItem>
`;
